library/src/main/java/com/novoda/downloadmanager/lib/DownloadStatus.java

```diff
@@ -0,0 +1,281 @@
+     * a content length but don't have one, and it is also used in the
+     * client when a response is received whose length cannot be determined
+     * accurately (therefore making it impossible to know when a download
+     * completes).
+     */
+    public static final int LENGTH_REQUIRED = 411;
+    /**
+     * This download was interrupted and cannot be resumed.
+     * This is the code for the HTTP error "Precondition Failed", and it is
+     * also used in situations where the client doesn't have an ETag at all.
+     */
+    public static final int PRECONDITION_FAILED = 412;
+    /**
+     * The lowest-valued error status that is not an actual HTTP status code.
+     */
+    public static final int MIN_ARTIFICIAL_ERROR_STATUS = 488;
+    /**
+     * The requested destination file already exists.
+     */
+    public static final int FILE_ALREADY_EXISTS_ERROR = 488;
+    /**
+     * Some possibly transient error occurred, but we can't resume the download.
+     */
+    public static final int CANNOT_RESUME = 489;
+    /**
+     * This download was canceled
+     */
+    public static final int CANCELED = 490;
+    /**
+     * This download has completed with an error.
+     * Warning: there will be other status values that indicate errors in
+     * the future. Use isStatusError() to capture the entire category.
+     */
+    public static final int UNKNOWN_ERROR = 491;
+    /**
+     * This download couldn't be completed because of a storage issue.
+     * Typically, that's because the filesystem is missing or full.
+     * Use the more specific {@link #INSUFFICIENT_SPACE_ERROR}
+     * and {@link #DEVICE_NOT_FOUND_ERROR} when appropriate.
+     */
+    public static final int FILE_ERROR = 492;
+    /**
+     * This download couldn't be completed because of an HTTP
+     * redirect response that the download manager couldn't
+     * handle.
+     */
+    public static final int UNHANDLED_REDIRECT = 493;
+    /**
+     * This download couldn't be completed because of an
+     * unspecified unhandled HTTP code.
+     */
+    public static final int UNHANDLED_HTTP_CODE = 494;
+    /**
+     * This download couldn't be completed because of an
+     * error receiving or processing data at the HTTP level.
+     */
+    public static final int HTTP_DATA_ERROR = 495;
+    /**
+     * This download couldn't be completed because of an
+     * HttpException while setting up the request.
+     */
+    public static final int HTTP_EXCEPTION = 496;
+    /**
+     * This download couldn't be completed because there were
+     * too many redirects.
+     */
+    public static final int TOO_MANY_REDIRECTS = 497;
+    /**
+     * This download couldn't be completed because another download in the batch failed.
+     */
+    public static final int BATCH_FAILED = 498;
+
+    /**
+     * This download has been submitted to the download executor but not yet started
+     */
+    public static final int SUBMITTED = 189;
```

Comment:
Yes but let's keep it at the top so it'll be easier to keep track of which numbers are  being used :) https://github.com/novoda/download-manager/commit/efcd3e0e2f24a7479044b34de1791ee2277bca41